File: src/Application Specific/Classes/Entity Collections/Jobs/EngineerVisit/EngineerVisitPartsAndProducts/EngineerVisitPartsAndProducts.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace EngineerVisitsPartsAndProducts

        Public Class EngineerVisitPartsAndProducts

            Private _dataTypeValidator As DataTypeValidator
            Public Sub New()
                _dataTypeValidator = New DataTypeValidator
            End Sub

#Region "Class Properties"

            Public Property IgnoreExceptionsOnSetMethods() As Boolean
                Get
                    Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
                End Get
                Set(ByVal Value As Boolean)
                    Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
                End Set
            End Property

            Public ReadOnly Property Errors() As Hashtable
                Get
                    Return _dataTypeValidator.Errors
                End Get
            End Property

            Public ReadOnly Property DTValidator() As DataTypeValidator
                Get
                    Return _dataTypeValidator
                End Get
            End Property

            Private _exists As Boolean = False
            Public Property Exists() As Boolean
                Get
                    Return _exists
                End Get
                Set(ByVal Value As Boolean)
                    _exists = Value
                End Set
            End Property

            Private _deleted As Boolean = False
            Public ReadOnly Property Deleted() As Boolean
                Get
                    Return _deleted
                End Get
            End Property
            Public WriteOnly Property SetDeleted() As Boolean
                Set(ByVal Value As Boolean)
                    _deleted = Value
                End Set
            End Property

#End Region

#Region "EngineerVisitPartsAndProducts Properties"

            Private _PartOrProductID As Integer = 0
            Public ReadOnly Property PartOrProductID() As Integer
                Get
                    Return _PartOrProductID
                End Get
            End Property
            Public WriteOnly Property SetPartOrProductID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_PartOrProductID", Value)
                End Set
            End Property

            Private _EngineerVisitID As Integer = 0
            Public ReadOnly Property EngineerVisitID() As Integer
                Get
                    Return _EngineerVisitID
                End Get
            End Property
            Public WriteOnly Property SetEngineerVisitID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_EngineerVisitID", Value)
                End Set
            End Property

            Private _Quantity As Integer = 0
            Public ReadOnly Property Quantity() As Integer
                Get
                    Return _Quantity
                End Get
            End Property
            Public WriteOnly Property SetQuantity() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Quantity", Value)
                End Set
            End Property

            Private _AssetID As Integer = 0
            Public ReadOnly Property AssetID() As Integer
                Get
                    Return _AssetID
                End Get
            End Property
            Public WriteOnly Property SetAssetID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_AssetID", Value)
                End Set
            End Property

            Private _AllocatedID As Integer = 0
            Public ReadOnly Property AllocatedID() As Integer
                Get
                    Return _AllocatedID
                End Get
            End Property
            Public WriteOnly Property SetAllocatedID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_AllocatedID", Value)
                End Set
            End Property

            Private _LocationID As Integer = 0
            Public ReadOnly Property LocationID() As Integer
                Get
                    Return _LocationID
                End Get
            End Property
            Public WriteOnly Property SetLocationID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_LocationID", Value)
                End Set
            End Property


            Private _SpecialPrice As Double = 0.0
            Public ReadOnly Property SpecialPrice() As Double
                Get
                    Return _SpecialPrice
                End Get
            End Property
            Public WriteOnly Property SetSpecialPrice() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_SpecialPrice", Value)
                End Set

            End Property

            Private _SpecialPartNumber As String = ""
            Public ReadOnly Property SpecialPartNumber() As String
                Get
                    Return _SpecialPartNumber
                End Get
            End Property
            Public WriteOnly Property SetSpecialPartNumber() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_SpecialPartNumber", Value)
                End Set
            End Property

            Private _SpecialDescription As String = ""
            Public ReadOnly Property SpecialDescription() As String
                Get
                    Return _SpecialDescription
                End Get
            End Property
            Public WriteOnly Property SetSpecialDescription() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_SpecialDescription", Value)
                End Set
            End Property



#End Region

        End Class

    End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Jobs/EngineerVisit/EngineerVisitPartsAndProducts/EngineerVisitPartsAndProducts.cs
using System.Collections;

namespace FSM.Entity
{
    namespace EngineerVisitsPartsAndProducts
    {
        public class EngineerVisitPartsAndProducts
        {
            private DataTypeValidator _dataTypeValidator;

            public EngineerVisitPartsAndProducts()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            
            
            private int _PartOrProductID = 0;

            public int PartOrProductID
            {
                get
                {
                    return _PartOrProductID;
                }
            }

            public object SetPartOrProductID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_PartOrProductID", value);
                }
            }

            private int _EngineerVisitID = 0;

            public int EngineerVisitID
            {
                get
                {
                    return _EngineerVisitID;
                }
            }

            public object SetEngineerVisitID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_EngineerVisitID", value);
                }
            }

            private int _Quantity = 0;

            public int Quantity
            {
                get
                {
                    return _Quantity;
                }
            }

            public object SetQuantity
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Quantity", value);
                }
            }

            private int _AssetID = 0;

            public int AssetID
            {
                get
                {
                    return _AssetID;
                }
            }

            public object SetAssetID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_AssetID", value);
                }
            }

            private int _AllocatedID = 0;

            public int AllocatedID
            {
                get
                {
                    return _AllocatedID;
                }
            }

            public object SetAllocatedID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_AllocatedID", value);
                }
            }

            private int _LocationID = 0;

            public int LocationID
            {
                get
                {
                    return _LocationID;
                }
            }

            public object SetLocationID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_LocationID", value);
                }
            }

            private double _SpecialPrice = 0.0;

            public double SpecialPrice
            {
                get
                {
                    return _SpecialPrice;
                }
            }

            public object SetSpecialPrice
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_SpecialPrice", value);
                }
            }

            private string _SpecialPartNumber = "";

            public string SpecialPartNumber
            {
                get
                {
                    return _SpecialPartNumber;
                }
            }

            public object SetSpecialPartNumber
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_SpecialPartNumber", value);
                }
            }

            private string _SpecialDescription = "";

            public string SpecialDescription
            {
                get
                {
                    return _SpecialDescription;
                }
            }

            public object SetSpecialDescription
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_SpecialDescription", value);
                }
            }



            
        }
    }
}